common/lib/dependabot/config/ignore_condition.rb

```diff
@@ -88,7 +89,7 @@ def ignore_patch(version)
       def ignore_minor(version)
         parts = version.split(".")
         version_parts = parts.fill("0", parts.length...2)
-        lower_parts = version_parts.first(1) + [version_parts[1].to_i + 1] + ["a"]
```

Comment:
I don't know for sure but I can only surmise that it's an attempt to include prereleases. In the parent class Gem::Version any version that contains an alphabetic letter is classed a prerelease. So 3.4.a  tries to include the earliest prereleases in the range.

Reply:
From my experience so far, I think the impact would be very small because most clients ignore prereleases, anyway. But then I don't have the stats to back this up.

Reply:
Thanks.

* How can we derisk this change? It's in common and will have a big blast radius - can it be scoped just to maven? Or can we use feature flags?
* How will this change when the Gem::Version is replaced with the new SemVer base you've introduced?

Reply:
[I just looked up history and this seems to the PR that introduced that change](https://github.com/dependabot/dependabot-core/pull/3588)

Reply:
It was introduced much earlier than that: https://github.com/dependabot/dependabot-core/pull/3513/files but there's no clear explanation for why `a` was chosen. The comments allude to what I explained above regarding compatibility with Gem::Version and versions with a letter being considered prereleases.

Reply:
> How can we de-risk this change? It's in common and will have a big blast radius - can it be scoped just to maven? Or can we use feature flags?

I could scope it to maven though adding these ecosystem checks in dependabot common looks like an antipattern. 

My preference is to use a feature flag because I'm sure assumptions like this one will not hold for other ecosystems with their own version identifier specifications too.

Reply:
Can we move the processing to `maven` like you did with versioning to guarantee minimal regression risk? A feature flag does not guarantee other things won't break. We have 26 ecosystems in total and a 1/26 understanding is high risk.

Reply:
> How will this change when the Gem::Version is replaced with the new SemVer base you've introduced?

Given my experience with this so far, I intend to have SemVer subclass Gem::Version and override all the methods that really matter. Completely removing Gem::Version at this point would just cause us endless pain given how tightly this code is coupled to Gem::Version